Extensions/ShareTo/InitialViewController.swift

```diff
@@ -3,134 +3,87 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import UIKit
+import SnapKit
+import Deferred
 import Shared
 import Storage
 
-private let LastUsedShareDestinationsKey = "LastUsedShareDestinations"
+/*
+ The initial view controller is full-screen and is the only one with a valid extension context.
+ It is just a wrapper with a semi-transparent background to darken the screen
+ that embeds the share view controller which is designed to look like a popup.
+
+ The share view controller is embedded using a navigation controller to get a nav bar
+ and standard iOS navigation behaviour.
+ */
+
+class EmbeddedNavController {
+    weak var parent: UIViewController?
+    var controllers = [UIViewController]()
+    var navigationController: UINavigationController
+
+    init (parent: UIViewController, rootViewController: UIViewController) {
+        self.parent = parent
+        navigationController = UINavigationController(rootViewController: rootViewController)
+
+        parent.addChildViewController(navigationController)
+        parent.view.addSubview(navigationController.view)
+
+        let width = min(UIScreen.main.bounds.width, CGFloat(UX.topViewWidth))
+
+        navigationController.view.snp.makeConstraints { make in
+            make.center.equalToSuperview()
+            make.width.equalTo(width)
+            make.height.equalTo(UX.topViewHeight)
+        }
+
+        navigationController.view.layer.cornerRadius = UX.dialogCornerRadius
+        navigationController.view.layer.masksToBounds = true
+    }
+
+    deinit {
+        navigationController.view.removeFromSuperview()
+        navigationController.removeFromParentViewController()
+    }
+}
 
 @objc(InitialViewController)
-class InitialViewController: UIViewController, ShareControllerDelegate {
-    var shareDialogController: ShareDialogController!
+class InitialViewController: UIViewController {
+    var embedController: EmbeddedNavController!
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        self.view.backgroundColor = UIColor(white: 0.0, alpha: 0.66) // TODO: Is the correct color documented somewhere?
+        view.backgroundColor = UIColor(white: 0.0, alpha: UX.alphaForFullscreenOverlay)
```

Comment:
BTW, the old value of 0.66 is too dark, I chose a value that appeared closer to the system's behaviour.